### PR TITLE
Kip 100

### PR DESCRIPTION
**Pull request by JaredBorders**

### CODE REVIEW

File: kips/kip-100.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+title: Smart Margin v3
+status: Draft
+created: 2023-08-1
+section: 4
+snapshot:
+---
+
+## Simple Summary
+
+Smart Margin v3 (SMv3) will serve as Kwenta's on-chain margin engine for trading derivatives powered by Synthetix v3. SMv3 will handle all trades created on the platform and serve as a single entry point for any on-chain integrators.
```
Comment (JChiaramonte7) [inline kips/kip-100.md:12]: Prefer `onchain` vs `on-chain`

File: kips/kip-100.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+```
+
+
+#### Conditional Orders
+
+https://github.com/Kwenta/smart-margin-v3/blob/main/diagrams/Conditional-Orders.png
+
+In SMv3, Conditional Orders gain enhanced functionality due to a restructured defining object. With order submission moving off-chain, the object can be more intricate without inflating gas costs (2). 
+
+Conditional Orders specify both order details (`IEngine.OrderDetails`) and on-chain verification criteria. This bundled data (`IEngine.ConditionalOrder`) is signed off-chain by the account owner and executed on-chain when conditions are met, either by a `trustedExecutor` or publicly if `requireVerified` is true.
```
Comment (JChiaramonte7) [inline kips/kip-100.md:82]: Would link to these contracts or paste struct interface here

File: kips/kip-100.md [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+SMv3 System Overview:
+
+https://github.com/Kwenta/smart-margin-v3/blob/main/diagrams/SMv3-Overview.png
+
+#### Engine
+
+The margin engine will interact with Synthetix v3 perp markets, guided by functions like `modifyCollateral` and `commitOrder` on the `Engine.sol` contract. With adequate interactions, the engine can completely manage users' Synthetix v3 perp market accounts (1). 
+
+Multi-leg interactions within a single transaction are possible, similar to SMv2, thanks to its multicallable feature. However, authorization is needed for the engine to access these accounts, which is granted by setting account permissions. These permissions are adjustable at any time, offering account creators the flexibility to include or exclude engines (4).
```
Comment (JChiaramonte7) [inline kips/kip-100.md:48]: Setting account permissions on Synthetix contracts or ours? I'm assuming theirs, but not clear here.

Also, is the margin engine itself something that needs to be granted permission to by the original user?

**Review by JChiaramonte7 — CHANGES_REQUESTED** [review]

File: kips/kip-100.md (hunk 2 above)
Comment (JaredBorders) [reply kips/kip-100.md]: Added struct

File: kips/kip-100.md (hunk 3 above)
Comment (JaredBorders) [reply kips/kip-100.md]: Added clarification

**Review by platschi — APPROVED** [review]

**Pull request merged**